tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/Simple.fs

```diff
@@ -126,4 +126,109 @@ match CaseA with
         |> withDiagnostics [
                 Warning 104, Line 10, Col 7, Line 10, Col 12, "Enums may take values outside known cases. For example, the value 'CaseB (enum<E> (0))' may indicate a case not covered by the pattern(s)."
                 Warning 25, Line 10, Col 7, Line 10, Col 12, "Incomplete pattern matches on this expression. For example, the value 'CaseA' may indicate a case not covered by the pattern(s)."]
-   
\ No newline at end of file
+   
+    [<Fact>]
+    let ``Pattern matching on a union case, single error, uses only pattern range`` () = 
+        Fsx """
+type Stuff =
+    | A
+    | B
+    | C
+
+let x v = 
+    match v with
+    | A -> ""
+    | A -> ""
+    | C -> ""
+        """
+        |> withOptions ["--test:ErrorRanges"]     
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            (Warning 25, Line 8, Col 11, Line 8, Col 12, "Incomplete pattern matches on this expression. For example, the value 'B' may indicate a case not covered by the pattern(s).")
+            (Warning 26, Line 10, Col 7, Line 10, Col 8, "This rule will never be matched")
+        ]
+    
+    [<Fact>]
+    let ``Pattern matching on a union case with when clause, single error, uses only pattern range`` () = 
+        Fsx """
+type Stuff =
+    | A
+    | B
+    | C
+
+let x v z = 
+    match v with
+    | A -> ""
+    | A when z > 5 -> ""
+    | C -> ""
+        """
+        |> withOptions ["--test:ErrorRanges"]     
+        |> typecheck
+        |> shouldFail
+        |> withDiagnostics [
+            (Warning 25, Line 8, Col 11, Line 8, Col 12, "Incomplete pattern matches on this expression. For example, the value 'B' may indicate a case not covered by the pattern(s).")
+            (Warning 26, Line 10, Col 7, Line 10, Col 8, "This rule will never be matched")
```

Comment:
Here I'd also rather have `A when z > 5`, as if - it would make more sense to me in an error.

Reply:
See my previous comment :)